tests/ember_debug/profile-manager-test.js

```diff
@@ -1,8 +1,203 @@
+      'template:simple',
+      hbs('Simple {{test-foo}} {{test-bar}}', {
+        moduleName: 'my-app/templates/simple.hbs',
+      })
+    );
+    this.owner.register(
+      'template:comments/index',
+      hbs('{{#each this.comments as |comment|}}{{comment}}{{/each}}', {
+        moduleName: 'my-app/templates/comments/index.hbs',
+      })
+    );
+    this.owner.register(
+      'template:posts',
+      hbs('Posts', { moduleName: 'my-app/templates/posts.hbs' })
+    );
+    this.owner.register(
+      'template:components/test-foo',
+      hbs('test-foo', {
+        moduleName: 'my-app/templates/components/test-foo.hbs',
+      })
+    );
+    this.owner.register(
+      'template:components/test-bar',
+      hbs(
+        '<!-- before --><div class="another-component"><span>test</span> <span class="bar-inner">bar</span></div><!-- after -->',
+        { moduleName: 'my-app/templates/components/test-bar.hbs' }
+      )
+    );
+  });
+
+  test('ProfileManager can be created', function (assert) {
+    const profileManager = new ProfileManager();
+
+    assert.ok(!!profileManager, 'it was created');
+    assert.equal(profileManager.profiles.length, 0, 'it has no profiles');
+  });
+
+  test('Highlights for render are shown as expected', async function (assert) {
+    await visit('/posts');
+
+    // enable highlight
+    run(() => EmberDebug.port.trigger('render:updateShouldHighlightRender', {
+      shouldHighlightRender: true,
+    }));
+
+    await visit('/simple');
+
+    const simpleComponent = find('.simple-component');
+    const anotherComponent = find('.another-component');
+
+    const [simpleHighlight, anotherHighlight] = document.getElementsByClassName(
+      'ember-inspector-render-highlight'
+    );
+
+    const simpleStyle = simpleHighlight.style;
+    const simpleExpected = simpleComponent.getBoundingClientRect();
+    assert.ok(simpleHighlight, 'simpleHighlight is visible');
+    assert.equal(getRounded(simpleStyle.left), getRounded(simpleExpected.x), 'same x as component');
+    assert.equal(getRounded(simpleStyle.top), getRounded(simpleExpected.y), 'same y as component');
+    assert.equal(getRounded(simpleStyle.width), getRounded(simpleExpected.width), 'same width as component');
+    assert.equal(getRounded(simpleStyle.height), getRounded(simpleExpected.height), 'same height as component');
+
+    const anotherStyle = anotherHighlight.style;
+
+    const anotherExpected = anotherComponent.getBoundingClientRect();
+    assert.ok(anotherHighlight, 'anotherHighlight is visible');
+    assert.equal(getRounded(anotherStyle.left), getRounded(anotherExpected.x), 'same x as component');
+    assert.equal(getRounded(anotherStyle.top), getRounded(anotherExpected.y), 'same y as component');
+    assert.equal(getRounded(anotherStyle.width), getRounded(anotherExpected.width), 'same width as component');
+    assert.equal(getRounded(anotherStyle.height), getRounded(anotherExpected.height), 'same height as component');
```

Comment:
could we add await for 1s to ensure DOM nodes is removed? and check it' using getElementsByClassName one more time

Reply:
not as easy as it sounds, but done.